src/orquestra/quantum/wip/operators/_pauli_operators.py

```diff
@@ -83,6 +84,34 @@ def _validate_type(object: Any) -> None:
         )
 
 
+def _parse_operator(op_str: str) -> Tuple[int, str]:
+    match = re.match(r"([XYZI])([0-9]+)", op_str, re.I)
+
+    if not match:
+        raise ValueError("Badly formatted string representation passed.")
+
+    return int(match.group(2)), match.group(1).upper()
+
+
+def _parse_operators_and_coefficient(
+    term_str: str,
+) -> Tuple[Optional[complex], Dict[int, str]]:
+    parts = term_str.split("*")
```

Comment:
I would strip the string of all spaces at the very beginning. Right now it errors out when I pass a string such as `PauliTerm("2 * Z1 * Z2")`

Reply:
It should no longer error out for strings with spaces inside them.